examples/neural_compressor/optical-character-recognition/run_ocr.py

```diff
@@ -0,0 +1,333 @@
+
+    model = VisionEncoderDecoderModel.from_pretrained(model_args.model_name_or_path)
+    model.config.decoder_start_token_id = processor.tokenizer.cls_token_id
+    model.config.pad_token_id = processor.tokenizer.pad_token_id
+    # make sure vocab size is set correctly
+    model.config.vocab_size = model.config.decoder.vocab_size
+
+    # # set beam search parameters
+    model.config.eos_token_id = processor.tokenizer.sep_token_id
+    model.config.encoder.num_beams = data_args.num_beams
+    model.to(device)
+
+    cer = load_metric("cer")
+
+    class AverageMeter(object):
+        """Computes and stores the average and current value"""
+
+        def __init__(self, name, fmt=":f"):
+            self.name = name
+            self.fmt = fmt
+            self.reset()
+
+        def reset(self):
+            self.val = 0
+            self.avg = 0
+            self.sum = 0
+            self.count = 0
+
+        def update(self, val, n=1):
+            self.val = val
+            self.sum += val * n
+            self.count += n
+            self.avg = self.sum / self.count
+
+        def __str__(self):
+            fmtstr = "{name} {val" + self.fmt + "} ({avg" + self.fmt + "})"
+            return fmtstr.format(**self.__dict__)
+
+    def eval_func(model, iters=None):
+        batch_time = AverageMeter("Time", ":6.3f")
+
+        for i, batch in enumerate(test_dataloader):
+            # predict using generate
+            pixel_values = batch["pixel_values"].to(device)
+            # setting channels_last
+            pixel_values = pixel_values.to(memory_format=torch.channels_last)
+            if i >= optim_args.warmup_iter:
+                start = time.time()
+            outputs = model.generate(pixel_values, num_beams=data_args.num_beams)
+            # measure elapsed time
+            if i >= optim_args.warmup_iter:
+                batch_time.update(time.time() - start)
+
+            # decode
+            pred_str = processor.batch_decode(outputs, skip_special_tokens=True)
+            labels = batch["labels"]
+            labels[labels == -100] = processor.tokenizer.pad_token_id
+            label_str = processor.batch_decode(labels, skip_special_tokens=True)
+
+            # add batch to metric
+            cer.add_batch(predictions=pred_str, references=label_str)
+            if iters is not None and i >= iters:
+                break
+        score = cer.compute()
+        print("Batch size = %d" % training_args.per_device_eval_batch_size)
+        if training_args.per_device_eval_batch_size == 1:
+            print("Latency: %.3f ms" % (batch_time.avg * 1000))
+        print("Throughput: %.3f images/sec" % (training_args.per_device_eval_batch_size / batch_time.avg))
+
+        # TODO: this should also be done with the ProgressMeter
+        print("cer: {score:.5f}".format(score=(score)))
+
+        return score
+
+    if optim_args.apply_quantization:
+        torch.backends.quantized.engine = "onednn"
```

Comment:
Could you remove this line in order for all users to be able to run this script ? We currently have no dependency to onednn

Reply:
removed